MARA-55 : 화면설계서 기반 냉장고, 친구 추가 API 개발, 리팩토링

jhkang1517 opened the pull request:
# Pull Request

## 작업 내용
이 PR은 다음과 같은 작업을 포함하고 있습니다:
- [x] 친구 냉장고 최신 근황 조회 
- [x] 친구 사용자 냉장고 중 가장 최근에 식자재가 추가된 냉장고의 식자재 리스트 조회
- [x] 친구 수 조회
- [x] 친구 삭제 기능은 기존 id 방식에서 list 방식으로 변경 적용
- [x] @Transactional 적용
- [x] 자신의 냉장고 리스트 조회

## 변경 사항
다음과 같은 파일/부분을 수정했습니다:
- `src/main/kotlin/mara/server/domain/friend/*.kt`: 친구 냉장고 관련 신규 컨트롤러, 서비스, DTO 추가
- `src/main/kotlin/mara/server/domain/refrigerator/*.kt`: 냉장고 조회 관련 수정
- - `src/main/kotlin/mara/server/domain/user/*.kt`: `UserNameResponse` -> `UserFriendResponse` 변경

## 관련 이슈
이 PR과 관련된 이슈: [MARA-55](https://alsehf0316.atlassian.net/browse/MARA-55)

## 테스트
다음과 같은 테스트를 수행했습니다:
- 단위 테스트: swagger 사용 단위 테스트 수행

[MARA-55]: https://alsehf0316.atlassian.net/browse/MARA-55?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jhkang1517 committed:
FriendshipService @Transactional 적용

jhkang1517 committed:
UserNameResponse -> UserFriendResponse 이름 변경

jhkang1517 committed:
친구 삭제 기능 FriendshipDeleteRequest -> List<FriendshipDeleteRequest> 변경

jhkang1517 committed:
친구 수 조회 기능 구현

jhkang1517 committed:
FriendRefrigerator Controller 껍데기 구현

jhkang1517 committed:
자신의 냉장고 리스트 조회 기능 구현

jhkang1517 committed:
냉장고 생성 기능 리팩토링 (로그인 한 사용자의 냉장고만 생성 가능하도록 변경)

jhkang1517 committed:
친구 냉장고 최신 근황 조회 기능 단순 구현

jhkang1517 committed:
main branch 반영

jhkang1517 committed:
친구 냉장고 최근 근황 조회 기능 개선

jhkang1517 committed:
친구 냉장고 최근 근황 조회 기능 오류 해결

jhkang1517 committed:
Pageable 적용

jhkang1517 committed:
ktlintFormat 적용

Jiwon-cho commented on src/main/kotlin/mara/server/domain/friend/FriendRefrigeratorDto.kt:
```diff
@@ -0,0 +1,35 @@
+import mara.server.domain.user.User
+import org.springframework.data.domain.Page
+
+data class FriendRefrigeratorDto(
```
쓰이는곳이 없는거 같은데 작성해둔 이유가 있을까요??

jhkang1517 replied:
초기에 사용 예정이였는데 현재는 더 이상 사용하지 않습니당.. 삭제했습니다! 🚫🚫

Jiwon-cho commented on src/main/kotlin/mara/server/domain/refrigerator/RefrigeratorRepository.kt:
```diff
@@ -7,4 +7,6 @@ import org.springframework.stereotype.Repository
 interface RefrigeratorRepository : JpaRepository<Refrigerator, Long> {
     fun findRefrigeratorsByUser(user: User): List<Refrigerator>
+
+    fun findRefrigeratorByUserInOrderByIngredientAddDateDesc(user: List<User>): List<Refrigerator>
```
기획 페이지로 봤을 때는' 냉장고에 식자재를 추가한 사용자의 최신 근황 리스트를 최대 5개 까지' 만 보여준 다고 되있는데, 이에 대한 부분이 빠져있는 것 같습니다.

jhkang1517 replied:
`userPageable`, `ingredientPageable` 로 나누어 적용 하였습니다! 🐬

Jiwon-cho commented on src/main/kotlin/mara/server/domain/friend/FriendshipService.kt:
```diff
@@ -29,31 +31,40 @@ class FriendshipService(
 
-        for (friendship: Friendship in friendshipList) {
-            friendshipRepository.delete(friendship)
+            for (friendship: Friendship in friendshipList) {
```
성능적으로는 크게 차이가 없을 수 있긴 하지만 ,비슷한 구문이 반복되서 ,

friendshipList.forEach { friendshipRepository.delete(it) }
 
이렇게 바꾸는것도 좋지 않을까 해서 적어봅니다.

jhkang1517 replied:
반영했습니다!! 🌈🌈

choipureum replied:
👏
https://github.com/Kotlin/kotlin-style-guide/issues/33

Jiwon-cho reviewed (changes requested):
리뷰 완료했습니다~!

jhkang1517 committed:
리뷰 사항 반영

1. FriendRefrigeratorDto 삭제

2. userPageable, ingredientPageable 적용
동일 Pageable 객체 사용으로 인해 @Qualifier 적용

3. for (friendship: Friendship in friendshipList) -> friendshipList.forEach { friendshipRepository.delete(it) } 변경

jhkang1517 committed:
ktlintFormat 적용

Jiwon-cho commented on src/main/kotlin/mara/server/domain/friend/FriendRefrigeratorController.kt:
```diff
@@ -0,0 +1,35 @@
+    fun getRecentFriendRefrigeratorList(
+        @Qualifier("userPageable")
+        @PageableDefault(
+            size = 5, sort = ["ingredientAddDate"], direction = Sort.Direction.DESC
```
Config 에서 따로 빈으로 등록해서 사용하면 디폴트는 필요없을꺼 같습니다.

jhkang1517 replied:
해당 내용 적용했습니다!! 💡💡

Jiwon-cho reviewed (commented):
재 리뷰 했습니다

jhkang1517 committed:
@PageableDefault 제거

Jiwon-cho reviewed (approved):
수고하셨습니다!
